### ISSUE TITLE
Provide easy access to the top of stdio of build step

### ISSUE THREAD

**Issue by mojca** [buildbot/buildbot#4855]
Our typical build logs may sometimes have hundreds of thousands of lines of output.

I really like how buildbot 2 quickly shows just the last few relevant lines which perfectly serves the purpose of both:
* knowing when you are in the build while still building / quickly inspecting what's going on
* getting you directly to the relevant error message responsible for the build failure
This work much much better than in 0.8.

However what I'm missing now is a quick access to the top of the file where all the relevant info about the build environment, flags etc. resides. At the moment I can only either download the full log or ask the browser to show me all the tens or hundreds of thousands of lines. And if I say "show all lines" during the build (even if that's at the beginning of the build), it's nearly useless because the new data arrives before I can even copy-paste something and then everything is purged and reloaded all the time.

**Comment by tardyp** [comment]
I don't understand, if you click on a log detail, it will show you the lines on demand, as you scroll.
It does not load the full log.
So you just have to scroll to the beginning.

We could easily add a scroll to top button aside from the scroll to bottom

![image](https://user-images.githubusercontent.com/109859/59910680-b21bcf80-9412-11e9-8fb9-a1f3039ddbfd.png)

**Comment by mojca** [comment]
OK, sorry, that was probably a wrong choice of words (it does allow scrolling rather than loading everything at once).

But as soon as I scroll at the beginning, I can only watch the contents for about a second or so, then it disappears and somehow ends up back at the end of the file. It works fine after the build, just not during the build.

Yes, "scroll to top" button would definitely help to some extent, but only if it didn't immediately scroll back to the bottom during the build.

**Comment by tardyp** [comment]
You have to uncheck the scroll to the end button. That button is a sticky push button to ensure the you follow the log when data arrives.
It is the default behaviour when the build is ungoing.

**Comment by rajdeepbharati** [comment]
scroll to top button can be added

**Comment by mojca** [comment]
Maybe then the main issue is in fact with how the info is represented to the user. Only after you explained that to me I realized that this was a button which was actually enabled (if it was a checkbox, I would see it; with a push button with hardly distinguishing shades of light gray I never noticed that the button was of a different colour than the others). I thought it meant "scroll to the bottom", but that it would only scroll back to the bottom from any other location if pressed. I didn't notice it was enabled (pressed down).

> scroll to top button can be added. It can be used after you uncheck the `scroll down` button.

I would invert the logic. I would put two buttons there: `scroll to top` & `scroll to bottom`, but as soon as you push `scroll to top`, the `scroll to bottom` should be immediately automatically deactivated. There's another thing. One first needs to press `show all N lines` to be even able to press any of those buttons. It would be nice to have `scroll to top` button available as soon as `stdio` (or any other log) is expanded, without having to click `load all N lines` before. It's ok now that I know it (I don't really need that button any longer now that I know how to operate it), but it's somewhat counterintuitive for a new users.

**Comment by dankegel** [comment]
There's a "scroll to end" button??!?   Wish I'd noticed that.
